cloudinit/config/cc_apk_configure.py

```diff
@@ -0,0 +1,236 @@
+                                The base URL of an Alpine repository, or
+                                mirror, to download official packages from.
+                                If not specified then it defaults to ``{}``
+                            """.format(DEFAULT_MIRROR))
+                        },
+                        'community_enabled': {
+                            'type': 'boolean',
+                            'default': False,
+                            'description': dedent("""\
+                                Whether to add the Community repo to the
+                                repositories file. By default the Community
+                                repo is not included.
+                            """)
+                        },
+                        'testing_enabled': {
+                            'type': 'boolean',
+                            'default': False,
+                            'description': dedent("""\
+                                Whether to add the Testing repo to the
+                                repositories file. By default the Testing
+                                repo is not included. It is only recommended
+                                to use the Testing repo on a machine running
+                                the ``Edge`` version of Alpine as packages
+                                installed from Testing may have dependancies
+                                that conflict with those in non-Edge Main or
+                                Community repos."
+                            """)
+                        },
+                        'version': {
+                            'type': 'string',
+                            'description': dedent("""\
+                                The Alpine version to use (e.g. ``v3.12`` or
+                                ``edge``)
+                            """)
+                        },
+                    },
+                    'required': ['version'],
+                    'minProperties': 1,
+                    'additionalProperties': False,
+                },
+                'local_repo_base_url': {
+                    'type': 'string',
+                    'description': dedent("""\
+                        The base URL of an Alpine repository containing
+                        unofficial packages
+                    """)
+                }
+            },
+            'required': [],
+            'minProperties': 1,  # Either preserve_repositories or alpine_repo
+            'additionalProperties': False,
+        }
+    }
+}
+
+__doc__ = get_schema_doc(schema)
+
+
+def handle(name, cfg, cloud, log, _args):
+    """
+    Call to handle apk_repos sections in cloud-config file.
+
+    @param name: The module name "apk-configure" from cloud.cfg
+    @param cfg: A nested dict containing the entire cloud config contents.
+    @param cloud: The L{CloudInit} object in use.
```

Comment:
I copied this terminology from cc_ca_certs.py and cc_resolv_conf.py.

Reply:
@raharper This is the only outstanding issue left.

Reply:
Turns it is http://epydoc.sourceforge.net/manual-epytext.html  markup to CrossReference Link.  I'd drop it since it's not in use; at some point we'll need to replace those with something but not today.